src/typechecker/FStar.TypeChecker.TcTerm.fs

```diff
@@ -561,7 +561,7 @@ and tc_maybe_toplevel_term env (e:term) : term                  (* type-checked
             (* We constrain this unification variable to be of the shape Type u *)
             (* for some new unification variable u *)
             let t, u = U.type_u () in
-            let g_opt = Rel.try_teq true env c'.res_typ t in
+            let g_opt = Rel.get_subtyping_prop env c'.res_typ t in
```

Comment:
I wonder why we don't just set the expected type to type_u in the environment and remove this case analysis

Reply:
Sure, but this also looks to be here only to compute an universe level. I think we should just call `env.universe_of`

Reply:
I made that change in f879a0bc3e6d1c3937a309caa3f875dab098ff8a, now in master.